codespan-reporting/src/term/views.rs

```diff
@@ -286,38 +293,72 @@ where
                 )?;
             }
 
-            let mut lines = labeled_file.lines.into_iter().peekable();
+            let mut lines = labeled_file
+                .lines
+                .iter()
+                .map(|(index, line)| {
+                    (
+                        index,
+                        line,
+                        line.multi_labels.iter().any(|l| {
+                            if let (_, _, MultiLabel::Left(true)) = l {
+                                false
+                            } else {
+                                true
+                            }
+                        }) || !line.single_labels.is_empty(),
```

Comment:
To be honest I'm finding it rather hard to understand what's going on here! Would be cool to see if we can figure out how to make this easier to understand, as the view/renderer code is already starting to get rather convoluted by my own hand. 😅